CRM/Civigiftaid/Utils/Contribution.php

```diff
@@ -95,19 +95,24 @@ public static function addContributionToBatch($contributionIDs, $batchID) {
     );
 
     foreach ($contributionIDs as $contributionID) {
-      //$batchContribution = new CRM_Core_DAO_EntityBatch( );
-      $batchContribution = new CRM_Batch_DAO_EntityBatch();
-      $batchContribution->entity_table = 'civicrm_contribution';
-      $batchContribution->entity_id = $contributionID;
-
       // check if the selected contribution id already in a batch
       // if not, add to batchContribution else keep the count of contributions that are not added
-
-      if ($batchContribution->find(TRUE)) {
+      $contributionNotInBatch = civicrm_api3('EntityBatch', 'get', [
+          'sequential' => 1,
+          'entity_table' => "civicrm_contribution",
+          'batch_id.type_id' => "giftaid_batch",
+          'entity_id' => $contributionID,
+      ]);
+
+      if ($contributionNotInBatch['count']) {
         $contributionsNotAdded[] = $contributionID;
         continue;
       }
 
+      $batchContribution =& new CRM_Batch_DAO_EntityBatch();
```

Comment:
No need to add the `&` ampersand here as objects are passed by reference by default anyways.